### PR TITLE
fix(useMediaQuery): Return boolean describing media query match on first render

### PR DESCRIPTION
**Pull request by ArttuOll**
### What is the current behavior, and the steps to reproduce the issue?

Described in #1000 .

### What is the expected behavior?

The value returned from the hook should be a boolean describing the media query match on first render instead of `undefined`.

### How does this PR fix the problem?
By initializing the state with the the boolean received from `matchMedia` and creating a `MediaQueryList` entry to the cache in the process.

[Here](https://codesandbox.io/s/jovial-yalow-rt5mn7?file=/src/App.tsx) is a CodeSandbox, which contains the example provided in #1000, but with the new implementation of `useMediaQuery`. Note that the animation is no longer triggered when the page loads.

**This is a breaking change, since the value returned on the first render is now different than before. (Returning `undefined` before was not a bug since it was documented)**

## Checklist

- [X] Have you read [contribution guideline](../../CONTRIBUTING.md)?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [X] Is there an existing issue for this PR?
  - #1000 
- [X] Have the files been linted and formatted?
- [X] Have the docs been updated to match the changes in the PR?
- [X] Have the tests been updated to match the changes in the PR?
- [X] Have you run the tests locally to confirm they pass?

### CODE REVIEW

**Comment by xobotyi** [comment]
One problem - it will cause hydration error for SSR.

Initialisation should be flagged via hook parameters.

**Comment by ArttuOll** [comment]
> One problem - it will cause hydration error for SSR.
> 
> Initialisation should be flagged via hook parameters.

Sounds doable. I'll implement this later.

**Comment by ArttuOll** [comment]
Btw, could you elaborate on how this causes an hydration error? I wasn't aware of this.

**Comment by xobotyi** [comment]
SSR will be first-rendered with undefined, client with some value - react will freak out on that.

**Comment by xobotyi** [comment]
Also I have an idea to replace all that crap with `initializeWithValue` props in favor of context provider that will automatically tell all our hooks that we're in SSR mode without the need of passing extra arguments to hooks

**Comment by ArttuOll** [comment]
> first render on server and in browser should match

Of course, makes sense! Thanks.

> Also I have an idea to replace all that crap with initializeWithValue props in favor of context provider that will automatically tell all our hooks that we're in SSR mode without the need of passing extra arguments to hooks

Interesting. Would the users then have to wrap their app in some sort of context provider?

**Comment by xobotyi** [comment]
Yes, they should have, but in my eyes it is way more convenient that to be sure that you pass proper hook config to each hook that has isomorphic behavior.

**Comment by ArttuOll** [comment]
@xobotyi How about something like I pushed in my last commit? I added `initializeWithValue` option and defaulted it to `true`, like with other hooks such as `useLocalStorageValue` etc.

File: src/useScreenOrientation/useScreenOrientation.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export type ScreenOrientation = 'portrait' | 'landscape';
  * hook uses CSS3 `orientation` media-query to check screen orientation.
  */
 export function useScreenOrientation(): ScreenOrientation | undefined {
-  const matches = useMediaQuery('(orientation: portrait)');
+  const matches = useMediaQuery('(orientation: portrait)', { initializeWithValue: false });
```
Comment (xobotyi) [inline src/useScreenOrientation/useScreenOrientation.ts:12]: Maybe we should also add an argument to this hook to yield correct value on init

**Review by xobotyi — CHANGES_REQUESTED** [review]

File: src/useScreenOrientation/useScreenOrientation.ts (hunk 1 above)
Comment (ArttuOll) [reply src/useScreenOrientation/useScreenOrientation.ts:12]: Good point. Did not even think about this hook too much 😄 I'll work on this tomorrow.

Comment (ArttuOll) [reply src/useScreenOrientation/useScreenOrientation.ts]: Done.

**Pull request merged**